doi-transformer/src/main/java/no/unit/nva/doi/transformer/utils/IsbnCleaner.java

```diff
@@ -0,0 +1,77 @@
+package no.unit.nva.doi.transformer.utils;
+
+
+import no.unit.nva.model.exceptions.InvalidIsbnException;
+import org.apache.commons.validator.routines.ISBNValidator;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.List;
+
+import static java.util.Objects.isNull;
+
+
+public final class IsbnCleaner {
+
+    public static final String INVALID_CHARS = "[^[0-9]Xx]+";
+    public static final String EMPTY_STRING = "";
+
+    private static final Logger logger = LoggerFactory.getLogger(IsbnCleaner.class);
+    public static final String ERROR_WHEN_TRYING_TO_CLEAN_ISSN = "Error when trying to clean ISBN: ";
+
+    private IsbnCleaner() {
+    }
+
+    /**
+     * Takes an input string that is an ISBN candidate, tests it and formats it if possible or returns null.
+     * @param value a string ISBN candidate.
+     * @return A string of a valid ISBN, or null.
+     */
+    public static String clean(String value) {
+
+        String isbnCandidate = stripSeparatorAndWhitespace(value);
+
+        if (isNull(isbnCandidate)) {
+            return null;
+        }
+
+        try {
+            checkIsbn(isbnCandidate);
+            return isbnCandidate;
+        } catch (InvalidIsbnException e) {
+            logger.warn(ERROR_WHEN_TRYING_TO_CLEAN_ISSN + e.getMessage());
+            return null;
+        }
+    }
+
+    /**
+     * This method removes potential formatting errors such as whitespace, non-hyphen separators,
+     * invalid characters and missing separators.
+     *
+     * @param value an ISSN candidate.
+     * @return a formatted ISSN candidate or null.
+     */
+    private static String stripSeparatorAndWhitespace(String value) {
+        if (isNull(value)) {
+            return null;
+        }
+
+        String cleanedString = StringUtils.removeMultipleWhiteSpaces(value).replaceAll(INVALID_CHARS, EMPTY_STRING);
```

Comment:
INVALID_CHARS…well, yes, the characters are to some extent not the core characters of ISBNS, but the legal separators are not technically invalid…if the ISBN passed to this method has been validated before being passed to this method, then it is just remove the unwanted separators, otherwise we could generate valid ISBNs from a random string :D